src/input/input_abstract.rs

```diff
@@ -276,6 +278,9 @@ impl<'py> ValidatedDict<'py> for Never {
     ) -> ValResult<R> {
         unreachable!()
     }
+    fn last_key(&self) -> Option<Self::Key<'_>> {
+        unreachable!()
```

Comment:
Now I look at this again, I think it's strictly more correct to have

```suggestion
        match *self {}
```

... can you maybe adjust here and the other `Never` trait implementations?

Reply:
I'll leave you to change this. What I've added is consistent with other methods in this implementation.